src/Cli/dotnet/commands/dotnet-workload/search/WorkloadSearchVersionsCommand.cs

```diff
@@ -123,5 +125,47 @@ public override int Execute()
 
             return 0;
         }
+
+        private List<string> GetVersions(int numberOfWorkloadSetsToTake)
+        {
+            var featureBand = new SdkFeatureBand(_sdkVersion);
+            var packageId = _installer.GetManifestPackageId(new ManifestId("Microsoft.NET.Workloads"), featureBand);
+
+            try
+            {
+                return PackageDownloader.GetLatestPackageVersions(packageId, numberOfWorkloadSetsToTake, packageSourceLocation: null, includePreview: _includePreviews)
+                    .GetAwaiter().GetResult()
+                    .Select(version => WorkloadSetVersion.FromWorkloadSetPackageVersion(featureBand, version.ToString()))
+                    .ToList();
+            }
+            catch (NuGetPackageNotFoundException)
+            {
+                Microsoft.DotNet.Cli.Utils.Reporter.Error.WriteLine(string.Format(LocalizableStrings.NoWorkloadVersionsFound, featureBand));
+                return null;
+            }
+        }
+
+        public string FindBestWorkloadSetFromComponents()
+        {
+            var versions = GetVersions(0);
```

Comment:
Does 0 here mean to get all the versions?  It would probably be good to clarify this.

Reply:
Yes. Do you think a comment would suffice, or how else would you recommend making it clearer?